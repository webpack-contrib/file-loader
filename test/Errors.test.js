import loader from '../src';

describe('Errors', () => {
  test('Loader Error', () => {
    const err = () => loader.call({ emitFile: false });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('Validation Error', () => {
    const err = () => loader.call({
      query: { useRelativePath: 1 },
      emitFile: true,
    });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });
});
