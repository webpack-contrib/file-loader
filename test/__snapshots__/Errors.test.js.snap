// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Loader Error 1`] = `
"File Loader

emitFile is required from module system"
`;

exports[`Errors Validation Error 1`] = `
"Validation Error

File Loader Invalid Options

options.useRelativePath should be boolean
"
`;
